### PR TITLE
Add a IsPressed property to Button

### PR DESCRIPTION
**Pull request by mterwoord**
Right now, it's not possible use the :pressed state for something else, like turning on a light-bulb (iot) for as long as the button is pressed. With this property, this is possible.

### CODE REVIEW

File: src/Avalonia.Controls/Button.cs [hunk 1]
```diff
@@ -227,6 +236,7 @@ protected override void OnPointerPressed(PointerPressedEventArgs e)
             {
                 PseudoClasses.Add(":pressed");
```
Comment (grokys) [inline src/Avalonia.Controls/Button.cs]: Now we have this property, we probably shouldn't be adding the `:pressed` pseudoclass ourselves. There's a `PseudoClass` method that can be used to make a pseudoclass shadow a property: https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Controls/Control.cs#L111

**Review by grokys — CHANGES_REQUESTED** [review]
Looks good! I know I questioned whether we should have an `IsPressed` property in gitter but I see that it exists in WPF so we should add it. Just one small change requested.

**Comment by mterwoord** [comment]
@grokys you had a chance to look at this again?

**Review by grokys — APPROVED** [review]

**Review by grokys — CHANGES_REQUESTED** [review]
Changes look good except CI is failing, possibly because of the change to the Portable.Xaml submodule? Did you intend to commit that change?

**Review by grokys — APPROVED** [review]
Hope you don't mind @mterwoord but I sorted out the submodule problems myself and updated from master, All looking good to merge now!

**Comment by mterwoord** [comment]
Sorry, didn't intend to change anything there..


2017-09-13 0:38 GMT+02:00 Steven Kirk <notifications@github.com>:

**Pull request merged**